google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -380,7 +384,63 @@ def limit num
 
           new_query.limit = Google::Protobuf::Int32Value.new value: num
 
-          Query.start new_query, parent_path, client
+          Query.start new_query, parent_path, client, limit_type: :first
+        end
+
+        ##
+        # Limits a query to return only the last matching documents.
+        #
+        # You must specify at least one "order by" clause for limitToLast queries.
+        # (See {#order}.)
+        #
+        # Results for `limit_to_last` queries are only available once all documents
+        # are received. Hence, `limit_to_last` queries cannot be streamed using
+        # {#listen}.
+        #
+        # @param [Integer] num The maximum number of results to return.
+        #
+        # @return [Query] New query with `limit_to_last` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.order(:name, :desc).limit_to_last(5)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def limit_to_last num
+          new_query = @query.dup
+
+          if new_query.nil? || new_query.order_by.nil? || new_query.order_by.empty?
+            raise "specify at least one order clause before calling limit_to_last"
+          end
+
+          if limit_type == :last # don't reverse order_by more than once
+            raise "limit_to_last may only be called once"
+          end
+
+          # Flip the orderBy directions since we want the last results
+          new_query.order_by.each do |order|
+            order.direction = order.direction.to_sym == :DESCENDING ? :ASCENDING : :DESCENDING
+          end
+
+          # Swap the cursors to match the flipped query ordering.
+          new_start_at = new_query.end_at.dup
+          new_end_at = new_query.start_at.dup
+          new_query.start_at = new_start_at
+          new_query.end_at = new_end_at
```

Comment:
You also need to switch the "before" direction.

Reply:
Fixed, thanks!